lib/Basic/LaneBasedExecutionQueue.cpp

```diff
@@ -190,6 +196,7 @@ class LaneBasedExecutionQueue : public ExecutionQueue {
   : ExecutionQueue(delegate), buildID(std::random_device()()), numLanes(numLanes),
         readyJobs(Scheduler::make(alg)), environment(environment)
   {
+    backgroundTaskMax = numLanes * 16;
```

Comment:
I have a question about this.  I'm working on [SR-10325](https://bugs.swift.org/browse/SR-10325?focusedCommentId=45543&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-45543) and I'm trying to understand where the total number of background tasks come from.  In particular, it seems like this is allowing 16 tasks _per lane_.  Is that correct?  Elsewhere in SPM, the number of lanes are set by the number of CPUs.  On an my 8-core system (Big+little, 4 each) SPM spawns enough processes to essentially DoS the system.

Reply:
This is indeed allowing 16 background tasks per lane. Note, however, that as far as I am aware, there are no native SwiftPM tasks that participate in the background task support of llbuild at this time. Build tsks that wish to use this feature must explicitly opt in to it by sending the control message to the build system and are expected to behave nicely (not use excessive CPU) once they have done so.

Reply:
Thanks!  I believe you're correct.  I set this to `numLanes * 1` and it behaved the same way.  It turns out that I need to control the number of lanes rather than the backgroundTaskMax.